### PR TITLE
Addresses stacked ensemble component CV for scikit-learn 0.24 support

### LINKED ISSUE #1592: Stacked ensembler: use same CV data splitter as the rest of automl
**Problem**
Currently, the stacked ensembler has its own setup for CV. `IterativeAlgorithm` calls [the `_make_stacked_ensembler` util](https://github.com/alteryx/evalml/blob/main/evalml/pipelines/utils.py#L221), but doesn't currently thread through the data splitter from automl search.

[The data splitter set up by default in the stacked ensembler](https://github.com/alteryx/evalml/blob/main/evalml/pipelines/components/ensemble/stacked_ensemble_base.py#L53) doesn't set `shuffle=True`, which [could lead to poor performance](https://github.com/alteryx/evalml/issues/1259) if the input dataset has an ordering. It also won't have the same settings for other parameters like `n_folds`, which isn't ideal.

Also, this difference [is preventing us from supporting sklearn 0.24.0](https://github.com/alteryx/evalml/pull/1591). Fixing this issue should allow us to support that version.

**Fix**
Let's have automl pass its data splitter through `IterativeAlgorithm` down into the stacked ensembler.

### PR DESCRIPTION
**Pull request by angela97lin**
Closes #1592

### CODE REVIEW

**Review by freddyaboulton — APPROVED** [review]
Looks good to me @angela97lin !
